router/utils_test.go

```diff
@@ -9,68 +9,98 @@ import (
 	"github.com/tendermint/tendermint/privval"
 )
 
-func randomRouterPorts() *config.Router {
+func randomRouterPorts() *cfg.Router {
 	port, err := tmnet.GetFreePort()
 	if err != nil {
 		panic(err)
 	}
-	return &config.Router{
+	return &cfg.Router{
 		JSONListenAddress: fmt.Sprintf("localhost:%d", port),
 		RESTListenAddress: fmt.Sprintf("localhost:%d", port+1),
 	}
 }
 
-func boostrapBVC(configfile string, workingdir string, baseport int) error {
-	idx := networks.IndexOf("Localhost")
-	net := networks.Networks[idx]
-	opts := node.InitOptions{}
+func initOptsForNetwork(t *testing.T, name string) node.InitOptions {
+	t.Helper()
+
+	// This will panic if `name` is wrong, causing the test to immediately fail
+	network := networks.Networks[networks.IndexOf(name)]
+	listenIP := make([]string, len(network.Nodes))
+	remoteIP := make([]string, len(network.Nodes))
+	config := make([]*cfg.Config, len(network.Nodes))
+
+	for i, net := range network.Nodes {
+		listenIP[i] = "tcp://0.0.0.0"
+		remoteIP[i] = net.IP
+		config[i] = new(cfg.Config)
+		config[i].Accumulate.Type = string(network.Type)
+
+		switch net.Type {
+		case cfg.Validator:
+			config[i].Config = *tmcfg.DefaultValidatorConfig()
+		case cfg.Follower:
+			config[i].Config = *tmcfg.DefaultValidatorConfig()
+		default:
+			fmt.Fprintf(os.Stderr, "Error: hard-coded network has invalid node type: %q\n", net.Type)
+			os.Exit(1)
+		}
+	}
+
+	return node.InitOptions{
+		ShardName: "accumulate.",
+		ChainID:   network.Name,
+		Port:      network.Port,
+		Config:    config,
+		RemoteIP:  remoteIP,
+		ListenIP:  listenIP,
+	}
+}
+
+func boostrapBVC(t *testing.T, configfile string, workingdir string) {
+	t.Helper()
+
+	opts := initOptsForNetwork(t, "Badlands")
 	opts.WorkDir = workingdir
-	opts.Port = net.Port
-	opts.ListenIP = make([]string, len(net.Nodes))
-	for i, node := range net.Nodes {
-		opts.ListenIP[i] = node.IP
+
+	for _, config := range opts.Config {
+		//[mempool]
+		//	broadcast = true
+		//	cache_size = 100000
+		//	max_batch_bytes = 10485760
+		//	max_tx_bytes = 1048576
+		//	max_txs_bytes = 1073741824
+		//	recheck = true
+		//	size = 50000
+		//	wal_dir = ""
+		//
+
+		// config.Mempool.KeepInvalidTxsInCache = false
+		// config.Mempool.MaxTxsBytes = 1073741824
+		config.Mempool.MaxBatchBytes = 1048576
+		config.Mempool.CacheSize = 1048576
+		config.Mempool.Size = 50000
 	}
-	node.Init(opts)
-
-	viper.SetConfigFile(configfile)
-	viper.AddConfigPath(filepath.Join(workingdir, "Node0"))
-	viper.ReadInConfig()
-	//[mempool]
-	//	broadcast = true
-	//	cache_size = 100000
-	//	max_batch_bytes = 10485760
-	//	max_tx_bytes = 1048576
-	//	max_txs_bytes = 1073741824
-	//	recheck = true
-	//	size = 50000
-	//	wal_dir = ""
-	//
-
-	//viper.Set("mempool.keep-invalid-txs-in-cache, "false"
-	//viper.Set("mempool.max_txs_bytes", "1073741824")
-	viper.Set("mempool.max_batch_bytes", 1048576)
-	viper.Set("mempool.cache_size", 1048576)
-	viper.Set("mempool.size", 50000)
-	err := viper.WriteConfig()
```

Comment:
`node.Init` writes out the config file, so all we need to do is set the properties on `node.Options.Config` prior to calling `node.Init`.